src/compiler/commandLineParser.ts

```diff
@@ -76,10 +76,11 @@ namespace ts {
                 "amd": ModuleKind.AMD,
                 "system": ModuleKind.System,
                 "umd": ModuleKind.UMD,
+                "es6": ModuleKind.ES6,

```

Comment:
we do not support `target: es5` with `module: es6`, so why allow it and add additional burden on users. 
`--target es6` with no `--module` is implied to be `es6`, if another `--module` tareget is specified, emit should use that.